digital-twin/e2e-tests/digital-twin-e2e-tests-common/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/e2e/simulator/TestInterfaceInstance2.java

```diff
@@ -83,10 +86,16 @@ public void onRegistered() {
     @Override
     protected void onPropertyUpdate(@NonNull DigitalTwinPropertyUpdate digitalTwinPropertyUpdate) {
         log.trace("Property received: name={}, desired value={}, version={}", digitalTwinPropertyUpdate.getPropertyName(), digitalTwinPropertyUpdate.getPropertyDesired(), digitalTwinPropertyUpdate.getDesiredVersion());
-        propertyUpdatesReceived.put(digitalTwinPropertyUpdate.getPropertyName(), digitalTwinPropertyUpdate.getPropertyDesired());
+        JsonNode desiredProperty = null;
+        try {
+            desiredProperty = objectMapper.readTree(digitalTwinPropertyUpdate.getPropertyDesired());
+        } catch (IOException e) {
+            log.error("Exception thrown while converting desired property update to Json Node: ", e);
+        }
+        propertyUpdatesReceived.put(digitalTwinPropertyUpdate.getPropertyName(), desiredProperty);
```

Comment:
move it up into `try` block

Reply:
Why would we need to wrap a `put` operation within a `try-catch` block?